src/Entity/PathautoPattern.php

```diff
@@ -0,0 +1,384 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\pathauto\Entity\PathautoPattern.
+ */
+
+namespace Drupal\pathauto\Entity;
+
+use Drupal\Core\Condition\ConditionPluginCollection;
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Plugin\Context\Context;
+use Drupal\Core\Plugin\Context\ContextDefinition;
+use Drupal\Core\Plugin\Context\ContextInterface;
+use Drupal\Core\Plugin\ContextAwarePluginInterface;
+use Drupal\Core\Plugin\DefaultSingleLazyPluginCollection;
+use Drupal\pathauto\PathautoPatternInterface;
+
+/**
+ * Defines the Pathauto pattern entity.
+ *
+ * @ConfigEntityType(
+ *   id = "pathauto_pattern",
+ *   label = @Translation("Pathauto pattern"),
+ *   handlers = {
+ *     "list_builder" = "Drupal\pathauto\PathautoPatternListBuilder",
+ *     "form" = {
+ *       "delete" = "Drupal\pathauto\Form\PathautoPatternDeleteForm"
+ *     },
+ *     "wizard" = {
+ *       "add" = "Drupal\pathauto\Wizard\PatternWizardAdd",
+ *       "edit" = "Drupal\pathauto\Wizard\PatternWizard"
+ *     }
+ *   },
+ *   config_prefix = "pathauto_pattern",
+ *   admin_permission = "administer site configuration",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "label" = "label",
+ *     "uuid" = "uuid"
+ *   },
+ *   links = {
+ *     "collection" = "/admin/config/search/path/patterns",
+ *     "edit-form" = "/admin/config/search/path/patterns/{machine_name}/{step}",
```

Comment:
I don't think this will work (slug name mismatch, so probably better to just leave it out? there might be code that expects it, though, we could try to make it work by overwriting urlRouteParameters()?